src/main/java/com/mixpanel/android/viewcrawler/ViewCrawler.java

```diff
@@ -175,18 +175,23 @@ public void onActivityCreated(Activity activity, Bundle bundle) {
 
         @Override
         public void onActivityStarted(Activity activity) {
+            ConnectivityIndicator.start(mContext);
+            mCurrentActivity = activity;
         }
 
         @Override
         public void onActivityResumed(Activity activity) {
             installConnectionSensor(activity);
             mEditState.add(activity);
+            mCurrentActivity = activity;
+            ConnectivityIndicator.show(activity);
```

Comment:
A view should not contain any app state. Calling `show` on any view should just show the view. Plus the state of this flag won't make the app to crash. We only use this flag to know whether the `ViewCrawler` should show this view or not. Hence, this should be outside the view and delegate this task to the `ViewCrawler`.